modules/subscription/main.tf

```diff
@@ -122,3 +122,26 @@ resource "azapi_resource_action" "subscription_cancel" {
   ]
 }
 
+resource "azapi_resource" "subscription_budget" {
+  for_each = var.subscription_budgets
+
+  type      = "Microsoft.Consumption/budgets@2021-10-01"
+  name      = each.key
+  parent_id = "/subscriptions/${local.subscription_id}"
+  body = jsonencode({
+    properties = {
+      amount        = each.value.amount
+      category      = "Cost"
+      notifications = each.value.notifications
```

Comment:
As noted in the previous comments, this object should be expanded in order to use snake case variables.

Reply:
added local to transform the notification map.